docs/book/src/developer/providers/v1.1-to-v1.2.md

```diff
@@ -55,20 +55,82 @@ in ClusterAPI are kept in sync with the versions used by `sigs.k8s.io/controller
   [merge-strategy](https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy).
   NOTE: the change will cause a rollout on existing clusters created with ClusterClass
 
-E.g. in CAPA
-
-```go
-// +optional
-Subnets Subnets `json:"subnets,omitempty"
-```
-Must be modified into:
-
-```go
-// +optional
-// +listType=map
-// +listMapKey=id
-Subnets Subnets `json:"subnets,omitempty"
-```
+  E.g. in CAPA
+
+  ```go
+  // +optional
+  Subnets Subnets `json:"subnets,omitempty"
+  ```
+  Must be modified into:
+
+  ```go
+  // +optional
+  // +listType=map
+  // +listMapKey=id
+  Subnets Subnets `json:"subnets,omitempty"
+  ```
+
+- [Server Side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) implementation in ClusterClass and managed topologies 
+  requires to dry-run changes on templates, and it is required to implement following changes in order to prevent dry-run
+  to return errors due to immutability checks implemented in the web hooks.
```

Comment:
```suggestion
  requires to dry-run changes on templates. If infrastructure or bootstrap providers have implemented immutability checks in their 
  InfrastructureMachineTemplate or BootstrapConfigTemplate webhooks, it is required to implement the following changes in order to prevent dry-run
  to return errors.
```

I think we have to add when it is required. Only a very small subset of infra providers have implemented the immutability check and it isn't documented / standardized anywhere as far as I can tell.

Q: Should we have a follow-up to document this somewhere in the contract? I think we can't expect folks to read through previous migration docs.

Reply:
I've adjusted the sentence but will leave this unresolved due to docs.

I'm +1 adding thsi to the contract.

Reply:
I think we can link to there but in the contract doc it's probably framed slightly different (not as a diff, but rather as what to do in ValidateUpdate if you want to do immutability checks)

Reply:
Maybe add a sentence like this:

```

Validating webhooks implemented for InfrastructureMachineTemplate or BootstrapConfigTemplate are required to not block due to immutability checks when the controller for managed topology and ClusterClass does
[Server Side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) dry-run requests.
```

to https://main.cluster-api.sigs.k8s.io/developer/providers/webhooks.html#validating-webhooks ? wdyt?

Reply:
That definitely doesn't hurt. I wonder if we should add a code snippet /example for ValidateUpdate + if it should be here instead: https://main.cluster-api.sigs.k8s.io/developer/providers/contracts.html

Reply:
With some restriction that it's only required if ClusterClass should be supported (similar to what we have here for using versions, using machines, ..): https://main.cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html#required-spec-fields-for-implementations-using-version

Probably makes sense to hear Fabrizio's take on it